### DISCUSSION TITLE
Trigger connected wallet and confirm transaction

### DISCUSSION THREAD

**Discussion by szanuje** [wevm/wagmi#426] [Q&A] [answered]
Hey, I'd like to call my contract's method which require sending some ether.

I assumed `useContractWrite` will do the work, but it actually doesn't seem to trigger my wallet, where MetaMask will estimate gas and let me confirm the transaction. Instead, it seem to me that it tries to send transaction in the background, but how would it even be possible without user's signature?

I'd like to trigger wallet extension and let me confirm and send the transaction to the contract.

```
  const { data: swapData, write: swap } = useContractWrite(
    {
      addressOrName: ethStorageAddress,
      contractInterface: ethStorageAbi,
      signerOrProvider: signer,
    },
    "swapExactETHForToken",
    {
      args: [],
      overrides: { value: ethers.utils.parseEther("0.001") },
    }
  );
```
```
 await swap();
```
I just get `cannot estimate gas` error.

Thanks!

**Comment by tmm** [comment, marked as answer]
MetaMask simulates the transaction before displaying a prompt to confirm. Likely there is an issue with the transaction you are trying to send.

**Reply by szanuje** [reply]
Thank you for quick answer. I reviewed my contract code once again and tested from Remix. It had a bug. The MetaMask prompt shows correctly now 👍
